posts/2024-03-29-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,161 @@
+
+== Getting your Cluster Ready
+
+In this blog, we assume you have some form of kubernetes cluster, which you have access to using the `kubectl` CLI or `oc` CLI if on OpenShift. 
+
+=== Using the Open Liberty Operator
+
+The Open Liberty Operator streamlines the deployment and management of Open Liberty applications on Kubernetes. Instead of manually crafting Kubernetes resources for the application (like services, deployments, routes, etc.), the Operator only requires us to define a few application details, and it automatically generates the required Kubernetes resources - saving us time and effort. Furthermore, it simplifies day 2 operations such as traces and heap dumps when needed. For more on the Open Liberty Operator check out the docs on it https://openliberty.io/docs/latest/open-liberty-operator.html[here].
+
+To take advantage of all of these benefits, we can make use of the Open Liberty Operator to deploy an application using the following commands, which install the operator with its default configuration (watching all namespaces):
+```
+git clone https://github.com/OpenLiberty/open-liberty-operator.git
+kubectl create namespace open-liberty
+kubectl apply --server-side -k open-liberty-operator/deploy/releases/1.2.2/kustomize/overlays/watch-all-namespaces
+```
+
+For a more detailed tutorial on the Open Liberty Operator and how to make use of it, see our interactive guide: https://openliberty.io/guides/openliberty-operator-intro.html[Deploying a microservice to Kubernetes using Open Liberty Operator].
+
+[#setup_argocd]
+=== Setup ArgoCD for effective GitOps and Continuous Deployment
+
+ArgoCD follows a GitOps pattern of using Git repositories as the source of truth for defining the desired application state, which we will use to deploy and monitor our Open Liberty configuration repository.  ArgoCD will help us identify any drift that occurs between the current cluster state and the desired state in the configuration repository.  
+
+To install ArgoCD, simply run the commands from the https://argo-cd.readthedocs.io/en/stable/getting_started/[Quickstart section] of the ArgoCD docs.
+
+
+First, we'll use the following commands to create a new namespace named `argocd` and install ArgoCD by applying manifests to that namespace.
+```
+kubectl create namespace argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+```
+
+Next, we'll need to get the default Admin credentials to access the ArgoCD user interface (UI).  Using the ArgoCD UI is a great way to get started, allowing us to create our application and identify configuration drift in a visual way.
+To get the default Admin password run the following command:
+```
+kubectl get secret argocd-initial-admin-secret -n argocd -o jsonpath='{.data.password}' | base64 -d
+```
+To access the ArgoCD UI, go to the route of the `argocd-server` and log with the `admin` username and password.
```

Comment:
just double checking that "route" is correct here vs "root", and if so, that it's clear how to go to the route